File: app/src/main/java/com/example/mvctutorial/designpattern/factory/DefaultItemCreator.java
package com.example.mvctutorial.designpattern.factory;


import com.example.mvctutorial.designpattern.factory.fw.Creator;
import com.example.mvctutorial.designpattern.factory.fw.Item;

public class DefaultItemCreator extends Creator {

    @Override
    protected String end(String itemName) {
        System.out.println("Default 마무리 작업!");
        return itemName;
    }

    @Override
    protected String init(String itemName) {
        System.out.println("Default 초기 작업!");
        return itemName;
    }

    @Override
    protected Item createItem(String itemName) {
        return new DefaultItem(itemName);
    }

}

File: app/src/main/java/com/example/mvctutorial/designpattern/_04factory/DefaultItemCreator.kt
package com.example.mvctutorial.designpattern._04factory

import com.example.mvctutorial.designpattern._04factory.fw.Creator
import com.example.mvctutorial.designpattern._04factory.fw.Item

class DefaultItemCreator : Creator() {

    override fun end(itemName: String): String {
        println("Default 마무리 작업!")
        return itemName
    }

    override fun init(itemName: String): String {
        println("Default 초기 작업!")
        return itemName
    }

    override fun createItem(itemName: String): Item {
        return DefaultItem(itemName)
    }
}